albumentations/augmentations/crops/functional.py

```diff
@@ -12,9 +12,9 @@
 
 
 def get_random_crop_coords(height: int, width: int, crop_height: int, crop_width: int, h_start: float, w_start: float):
-    y1 = int((height - crop_height) * h_start)
+    y1 = int((height - crop_height + 1) * h_start)
```

Comment:
I think better just change `int` to `round` because this code is a little bit strange for me:
```
y1 = round((height - crop_height) * h_start)
```

Reply:
It would also be great to add a comment here on why we're using this approach so that we don't forget about these changes.

Reply:
`int` implicitly creates the correct boundaries. Your provided example doesn't give the same boundaries/answer. e.g. `height=9` `crop_height=4` and `h_start=0.1001`

Let me try to explain why `int` is better via visualisation. I'm having trouble explaining myself in words only. :frowning_face: 

![sample-range](https://user-images.githubusercontent.com/10515040/146157931-c492126c-0443-459a-b321-997f3120426b.png)

The largest index in this theoretical image is 8, there are 9 pixels on the number line, and 10 vertical notches on the number line. Surprisingly tricky, right?

Let's say we have an image with 9 pixels, and we're taking a crop 4 pixels large. It should be clear from the visualisation that the last index we can use is 5 (height - crop_height). Then, when defining what range `h_start` maps onto, I've colour-coded the regions of the number line which map to each index for each solution. And `h_start` is the proportion of the rainbow. You see that the `+1` increases the width on the number line to fit in the last index.

Now, I just need to put this into a small comment that doesn't confuse things further... Before I put in a commit, can I get you to check whether this comment makes sense?

```python
# h_start is [0, 1) and should map to [0, (height - crop_height)] (note inclusive)
# this is conceptually equivalent to mapping onto `range(0, (height - crop_height + 1))`
```

Reply:
It is really tricky, but the current solution also doesn't work correctly, because if you try your code from your issue with these changes it will sometimes produce an empty image.

Reply:
Thanks for explanation, it is really tricky.
I think it would also be great if you provided a link to this PR in the comment.